src/components/RegistrationFlow/Picture/index.js

```diff
@@ -50,18 +75,83 @@ const Picture = props => {
         </div>
 
         <div className="add-user-picture-content-container">
-          <Avatar
-            width={115}
-            height={115}
-            label="&nbsp;"
-            labelStyle={customLabelStyle}
-            borderStyle={customBorderStyle}
-            onBeforeFileLoad={onBeforeFileLoad}
-            onCrop={onChange}
-            className="add-user-picture-picker"
-          />
+          {showFileLoader ? (
+            <ImageUploader
+              onChange={setImageFile}
+              setShowFileLoader={setShowFileLoader}
+            />
+          ) : (
+            <div className="image-preview-container">
+              <ButtonContainer
+                className="icon-btn close-preview-button"
+                onClick={closePreview}
+              >
+                <i className="fas fa-times" />
+              </ButtonContainer>
+              <AvatarEditor
+                ref={editorRef}
+                image={imageFile}
+                width={200}
+                height={200}
+                border={50}
+                borderRadius={100}
+                color={[255, 255, 255, 0.8]} // RGBA
+                scale={parseFloat(scale)}
+                rotate={parseFloat(rotation)}
+                position={position}
+                onPositionChange={handlePositionChange}
+                className="add-user-picture-picker"
+                onImageReady={savePreview}
```

Comment:
when is the `onImageReady` event fired? I was just thinking that do we even need savePreview here or does the useEffect (onImageChange) cover that already?

Reply:
Good point! onImageReady is necessary because if the user only uploads the image and doesn't rotate, scale or zoom, the image would not be saved. However, I noticed that in the current implementation it wasn't even working, and realised that I could do this better with a boolean in state. What do you think? I also removed the onimageChange since that's not really necessary either, as far as I can see..

Reply:
Ok cool! Current implementation seems to work as intended! :)